### PR TITLE
Fix: update latest version Fix: 1.2 upgrade 1.3 workflowstep XXX not found

### PR DESCRIPTION
**Pull request by cezhang**
Signed-off-by: cezhang <c1zhang.dev@gmail.com>

handle publishversion case

Signed-off-by: cezhang <c1zhang.dev@gmail.com>

### Description of your changes

Fixes "1.2 upgrade 1.3 workflowstep XXX not found" issue

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested
- vela install --version 1.2.0 in k3d
- create application through CLI(explicitly define workflow) and velaux
- vela install --version 1.3.3 in k3d
- watch application running from rendering to running
- check the apprev if workstepdefinitions are patched

### CODE REVIEW

**Comment by fourierr** [comment]
LGTM

File: pkg/appfile/parser.go [hunk 1]
```diff
@@ -288,6 +288,29 @@ func (p *Parser) GenerateAppFileFromRevision(appRev *v1beta1.ApplicationRevision
 	for k, v := range appRev.Spec.WorkflowStepDefinitions {
 		appfile.RelatedWorkflowStepDefinitions[k] = v.DeepCopy()
 	}
+
+	if len(appfile.RelatedWorkflowStepDefinitions) == 0 && len(appfile.WorkflowSteps) > 0 {
```
Comment (wonderflow) [inline pkg/appfile/parser.go:292]:
```suggestion
        // add compatible code for upgrading to v1.3 as the workflow steps were not recorded before v1.2
	if len(appfile.RelatedWorkflowStepDefinitions) == 0 && len(appfile.WorkflowSteps) > 0 {
```

**Review by wonderflow — COMMENTED** [review]
great job, let's add some test cases

**Review by wonderflow — APPROVED** [review]
great job, lgtm after nits, please add the explanation on my comments

File: pkg/appfile/parser_test.go [hunk 2]
```diff
@@ -506,3 +506,473 @@ var _ = Describe("Test appFile parser", func() {
...
+
+var _ = Describe("Test application parser", func() {
+	var app v1beta1.Application
+	var apprev v1beta1.ApplicationRevision
+	var wsd v1beta1.WorkflowStepDefinition
+	var expectedExceptAppfile *Appfile
+
+	BeforeEach(func() {
+		// prepare application
+		appYaml := `
```
Comment (Somefive) [inline pkg/appfile/parser_test.go:518]: Recommend moving these YAML configurations into testdata/app.yaml(apprev.yaml). They are too large.

**Review by Somefive — COMMENTED** [review]
Generally LGTM after nit.

**Pull request merged**